### PR TITLE
generate aggregation results for COG and Pfam. Add tests

### PR DESCRIPTION
**Pull request by chienchi**
* generate aggregation results for COG and Pfam by parsing corresponding line in the gff file.
* rename get_kegg_counts  to get_functional_annotation_counts
* add pytest on AnnotationLine Class and get_functional_annotation_counts in MetaGenomeFuncAgg Class

### CODE REVIEW

**Review by aclum — APPROVED** [review]
Looks great, thanks for adding the tests as well.

File: tests/test_generate_functional_agg.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+def test_AnnotationLine():
+    line = "nmdc:wfmtan-11-5rqhd817.1_0000001	Prodigal v2.6.3_patched	CDS	2931	5588	340.0	+	0	ID=nmdc:wfmgan-11-5rqhd817.1_0000001_2931_5588;translation_table=11;start_type=ATG;product=O-antigen biosynthesis protein;product_source=KO:K20444;cath_funfam=3.20.20.80,3.90.550.10;cog=COG0463;ko=KO:K20444;ec_number=EC:2.4.1.-;pfam=PF00535,PF02836;superfamily=51445,53448"
+    anno = AnnotationLine(line)
+    assert anno
+    assert len(anno.kegg) > 0
+    assert anno.id == "nmdc:wfmgan-11-5rqhd817.1_0000001_2931_5588"
+    assert anno.cogs == ["COG:COG0463"]
+    assert anno.kegg == ["KEGG.ORTHOLOGY:K20444"]
+    assert anno.pfams == ["PFAM:PF00535","PFAM:PF02836"]
```
Comment (eecavanna) [inline tests/test_generate_functional_agg.py:13]:
```suggestion
    assert anno.pfams == ["PFAM:PF00535", "PFAM:PF02836"]
```

**Review by eecavanna — APPROVED** [review]
Looks good to me (in terms of modifying existing Python code). I am not familiar with the algorithm, itself, and so don't have an opinion on the changes to that.

I left one suggestion about formatting array elements.

Thanks for adding the tests! Will you add a blurb to the `README.md` file, about how someone can run those tests? If you want to jot down a couple lines here in the PR (as a comment), I can take care of adding them to the `README.md` file.

**Review by eecavanna — APPROVED** [review]
The updates to `README.md` and `requirements.txt` look good to me — thanks! I'm OK with this branch being merged in as is.

**Pull request merged**